src/high_level.rs

```diff
@@ -554,3 +554,66 @@ pub fn find_cell_by_data_hash(data_hash: &[u8], source: Source) -> Result<Option
     }
     Ok(None)
 }
+
+/// Look for a dep cell with specific code hash, code_hash should be a buffer
+/// with 32 bytes.
+///
+pub fn look_for_dep_with_hash2(
+    code_hash: &[u8],
+    hash_type: ScriptHashType,
+) -> Result<usize, SysError> {
+    let field = match hash_type {
+        ScriptHashType::Type => CellField::TypeHash,
+        _ => CellField::DataHash,
+    };
+    let mut current: usize = 0;
+    loop {
+        let mut buf = [0u8; 32];
+        match syscalls::load_cell_by_field(&mut buf, 0, current, Source::CellDep, field) {
+            Ok(len) => {
+                debug_assert_eq!(len, buf.len());
+                if buf == code_hash {
+                    return Ok(current);
+                }
+            }
+            Err(SysError::ItemMissing) => {}
+            Err(SysError::IndexOutOfBound) => {
+                return Err(SysError::IndexOutOfBound);
+            }
+            Err(err) => {
+                return Err(err);
+            }
+        }
+        current += 1;
+    }
+}
+
+pub fn look_for_dep_with_data_hash(data_hash: &[u8]) -> Result<usize, SysError> {
+    look_for_dep_with_hash2(data_hash, ScriptHashType::Data)
+}
+
+/// Exec a cell in cell dep.
+///
+/// # Arguments
+///
+/// * `code_hash` - the code hash to search cell in cell deps.
+/// * `hash_type` - the hash type to search cell in cell deps.
+pub fn exec_cell(
+    code_hash: &[u8],
+    hash_type: ScriptHashType,
+    offset: u32,
+    length: u32,
+    argc: i32,
+    argv: &[&CStr],
+) -> Result<u64, SysError> {
+    let index = look_for_dep_with_hash2(code_hash, hash_type)?;
+    let bounds: usize = (offset as usize) << 32 | (length as usize);
+    Ok(syscalls::exec(
+        index,
+        Source::CellDep,
+        0,
+        bounds,
+        argc,
+        argv,
```

Comment:
The last element in argv should be 0. For example, when argc is 2, then argv[0], argv[1] are for arguments, argv[2] should be zero.

Reply:
`argv` is a `CStr` array, when argc is 2, the content of `argv` may like:
- `argv[0]` = "abc\0"
- `argv[1]` = "xyzijk\n"

And there is no argv[2].

Reply:
Yes you are right, the argv[argc] is null pointer.

https://www.gnu.org/software/libc/manual/html_node/Program-Arguments.html